IdentityCore/src/webview/embeddedWebview/challangeHandlers/MSIDClientTLSHandler.m

```diff
@@ -0,0 +1,229 @@
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import "MSIDChallengeHandler.h"
+#import "MSIDClientTLSHandler.h"
+#import "MSIDWorkPlaceJoinUtil.h"
+#import "MSIDRegistrationInformation.h"
+#import "MSIDWorkPlaceJoinConstants.h"
+#import "MSIDCertificateChooser.h"
+
+@implementation MSIDTLSHandler
+
++ (void)load
+{
+    [MSIDChallengeHandler registerHandler:self authMethod:NSURLAuthenticationMethodClientCertificate];
+}
+
++ (void)resetHandler { }
+
++ (BOOL)handleChallenge:(NSURLAuthenticationChallenge *)challenge
+                webview:(WKWebView *)webview
+                context:(id<MSIDRequestContext>)context
+      completionHandler:(ChallengeCompletionHandler)completionHandler
+{
+    NSString *host = challenge.protectionSpace.host;
+    
+    MSID_LOG_INFO(context, @"Attempting to handle client TLS challenge");
+    MSID_LOG_INFO_PII(context, @"Attempting to handle client TLS challenge. host: %@", host);
+    
+    // See if this is a challenge for the WPJ cert.
+    NSArray<NSData*> *distinguishedNames = challenge.protectionSpace.distinguishedNames;
+    if ([self isWPJChallenge:distinguishedNames])
+    {
+        return [self handleWPJChallenge:challenge context:context completionHandler:completionHandler];
+    }
+#if TARGET_OS_IPHONE
+    completionHandler(NSURLSessionAuthChallengePerformDefaultHandling, nil);
+    return NO;
+#else
+    return [self handleCertAuthChallenge:challenge webview:webview context:context completionHandler:completionHandler];
+#endif
+}
+
++ (BOOL)isWPJChallenge:(NSArray *)distinguishedNames
+{
+    
+    for (NSData *distinguishedName in distinguishedNames)
+    {
+        NSString *distinguishedNameString = [[[NSString alloc] initWithData:distinguishedName encoding:NSISOLatin1StringEncoding] lowercaseString];
+        if ([distinguishedNameString containsString:[kMSIDProtectionSpaceDistinguishedName lowercaseString]])
+        {
+            return YES;
+        }
+    }
+    
+    return NO;
+}
+
++ (BOOL)handleWPJChallenge:(NSURLAuthenticationChallenge *)challenge
+                   context:(id<MSIDRequestContext>)context
+         completionHandler:(ChallengeCompletionHandler)completionHandler
+{
+    NSError *localError = nil;
```

Comment:
this localError is being used anywhere?

Reply:
nope, removed and pass in nil instead :)